app/src/main/java/unipd/se18/ocrcamera/ManualTestOnSinglePhoto.java

```diff
@@ -0,0 +1,212 @@
+package unipd.se18.ocrcamera;
+
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.Matrix;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import info.debatty.java.stringsimilarity.Damerau;
+import unipd.se18.ocrcamera.recognizer.OCR;
+import unipd.se18.ocrcamera.recognizer.OCRListener;
+import unipd.se18.ocrcamera.recognizer.TextRecognizer;
+
+import static unipd.se18.ocrcamera.recognizer.TextRecognizer.getTextRecognizer;
+
+/**
+ * Class that compare the image stored image in preferences with the same image rotate
+ * of a certain angle given by the user, showing the similarity
+ */
+
+public class ManualTestOnSinglePhoto extends AppCompatActivity {
+
+    private String startingOCRText;
+
+    /**
+     * Modification variable
+     */
+    private int angleRotation;
+
+    /**
+     * UI
+     */
+    private EditText degreeEditText;
```

Comment:
degreeEditText is used only in onCreate. Is there a reason to keep it outside the onCreate?

Reply:
Just to have all the variable at the beginning  of the class. Do u think it's better that way or create a new variable in the method where it is used, if it's used only in this method?

Reply:
Better to create a variable inside the method if its used only there